roles/openshift_hosted/tasks/registry/secure.yml

```diff
@@ -2,21 +2,43 @@
 - name: Set fact docker_registry_route_hostname
   set_fact:
     docker_registry_route_hostname: "{{ 'docker-registry-default.' ~ (openshift_master_default_subdomain | default('router.default.svc.cluster.local', true)) }}"
+  run_once: true
+
+- name: Get the certificate contents for registry
+  copy:
+    backup: True
+    dest: "/etc/origin/master/named_certificates/{{ item | basename }}"
+    src: "{{ item }}"
+  register: openshift_hosted_registry_certificate_content
+  with_items:
+  - "{{ (openshift_hosted_registry_route_certificates | default({'certfile':none})).certfile }}"
+  - "{{ (openshift_hosted_registry_route_certificates | default({'keyfile':none})).keyfile }}"
+  - "{{ (openshift_hosted_registry_route_certificates | default({'cafile':none})).cafile }}"
+  when: openshift_hosted_registry_route_certificates
```

Comment:
Instead of defaulting `openshift_hosted_registry_route_certificates` to `false`, you can just default it to:
```
openshift_hosted_registry_route_certificates:
  certfile: None
  keyfile: None
  cafile: None
```
and then update this task to:
```
- name: Get the certificate contents for registry
  copy:
    backup: True
    dest: "/etc/origin/master/named_certificates/{{ item.value | basename }}"
    src: "{{ item.value }}"
  register: openshift_hosted_registry_certificate_content
  when: item.key in ['certfile', 'keyfile', 'cafile'] and item.value is not None
  with_dict: "{{ openshift_hosted_registry_route_certificates }}"
```